src/libsyntax/lib.rs

```diff
@@ -18,6 +18,7 @@
 #![feature(step_trait)]
 #![feature(try_trait)]
 #![feature(unicode_internals)]
+#![feature(rustc_private)]
```

Comment:
The changes from the first commit (so `cargo check` works) were of drive-by nature, can remove them if needed

Reply:
Yes, please remove.

Reply:
Done. Out of curiosity, why should  we skip this? Is it because it adds noise and is somewhat obvious? (after all we're in *the* compiler code)

Reply:
Less noise and impossible to forget something if it's added automatically.
rustc crates are not supposed to be built in isolation, outside of the rustc build system, so boilerplate common for all rustc crates is regularly moved to the workspace or rustbuild.
The `rustc_private` features in particular were moved in https://github.com/rust-lang/rust/pull/41847.

Reply:
IIRC, `x.py check` should provide the same functionality as `cargo check`.